src/Database/LSMTree/Internal/IndexOrdinary.hs

```diff
@@ -72,15 +72,19 @@ search key (IndexOrdinary lastKeys)
 
     start :: Int
     start | protoStart < pageCount = protoStart
-          | otherwise              = maybe 0 succ $
+          | otherwise              = maybe 0 succ                              $
                                      findIndexR (/= lastKeys ! pred pageCount) $
                                      lastKeys
 
     end :: Int
-    end | protoStart < pageCount
-            = start + length (takeWhile (== key) (drop (succ start) lastKeys))
-        | otherwise
-            = pred pageCount
+    end | protoStart < pageCount = start + overflowPageCount
+        | otherwise              = pred pageCount
+        where
+
+        overflowPageCount :: Int
+        overflowPageCount = length                          $
+                            takeWhile (== lastKeys ! start) $
+                            drop (succ start) lastKeys
```

Comment:
If we use `findIndex`, we avoid relying on the optimiser to get rid of the intermediate vector. We also use `findIndexR` above to do something very similar, so it would be consistent.

Reply:
Yes, this makes a lot of sense. Done in 1c036b3c2b793fc0c535558077800f55f4e0fd24.

Reply:
Absolutely. Done as part of 9486ca27fdb2f335729f5f6f1d50982022e306ec.